pudb/shell.py

```diff
@@ -249,7 +249,7 @@ class DummyMod:
 def run_ptpython_shell(globals, locals):
     # Use the default ptpython history
     import os
-    history_filename = os.path.expanduser("~/.ptpython/history")
```

Comment:
This is still supported as the "legacy history file":
https://github.com/prompt-toolkit/ptpython/blob/d8b5c90d20cc1e8d88837aa7163289950c231f74/ptpython/entry_points/run_ptpython.py#L131

Ideally, we would look up the history file using ptpython's methods, or at least emulate them or allow users to configure it to the desired location.

Reply:
How do you think of `os.access(os.path.expanduser("~/.ptpython/history"), os.W_OK)`?

Reply:
If `~/.ptpython/history` don't exist, let us use `~/.local/share/ptpython/history`.

Reply:
Why not just call `ptpython.get_config_and_history_file`, as @mvanderkamp suggests?

https://github.com/prompt-toolkit/ptpython/blob/d8b5c90d20cc1e8d88837aa7163289950c231f74/ptpython/entry_points/run_ptpython.py#L131

Reply:
Thanks! Failed the linter, otherwise LGTM!